jerry-core/jerry.c

```diff
@@ -119,526 +124,565 @@ jerry_make_api_available (void)
-} /* jerry_create_string */
+  return ecma_op_is_callable (value);
+} /* jerry_value_is_function */
 
 /**
- * Create a string
+ * Check if the specified value is number.
  *
- * Note:
- *      caller should release the string with jerry_release_string, just when the value becomes unnecessary.
- *
- * @return pointer to created string
+ * @return true  - if the specified value is number,
+ *         false - otherwise.
  */
-jerry_string_t *
-jerry_create_string_sz (const jerry_char_t *v, /**< string value */
-                        jerry_size_t v_size) /**< string size */
+bool
+jerry_value_is_number (const jerry_value_t value) /**< api value */
 {
-  jerry_assert_api_available ();
-
-  return ecma_new_ecma_string_from_utf8 ((lit_utf8_byte_t *) v,
-                                         (lit_utf8_size_t) v_size);
-} /* jerry_create_string_sz */
+  return ecma_is_value_number (value);
+} /* jerry_value_is_number */
 
 /**
- * Create an object
+ * Check if the specified value is null.
  *
- * Note:
- *      caller should release the object with jerry_release_object, just when the value becomes unnecessary.
+ * @return true  - if the specified value is null,
+ *         false - otherwise.
+ */
+bool
+jerry_value_is_null (const jerry_value_t value) /**< api value */
+{
+  return ecma_is_value_null (value);
+} /* jerry_value_is_null */
+
+/**
+ * Check if the specified value is object.
  *
- * @return pointer to created object
+ * @return true  - if the specified value is object,
+ *         false - otherwise.
  */
-jerry_object_t *
-jerry_create_object (void)
+bool
+jerry_value_is_object (const jerry_value_t value) /**< api value */
 {
-  jerry_assert_api_available ();
+  return ecma_is_value_object (value);
+} /* jerry_value_is_object */
 
-  return ecma_op_create_object_object_noarg ();
-} /* jerry_create_object */
+/**
+ * Check if the specified value is string.
+ *
+ * @return true  - if the specified value is string,
+ *         false - otherwise.
+ */
+bool
+jerry_value_is_string (const jerry_value_t value) /**< api value */
+{
+  return ecma_is_value_string (value);
+} /* jerry_value_is_string */
 
 /**
- * Create an array object
+ * Check if the specified value is undefined.
  *
- * Note:
- *      caller should release the object with jerry_release_object, just when the value becomes unnecessary.
+ * @return true  - if the specified value is undefined,
+ *         false - otherwise.
+ */
+bool
+jerry_value_is_undefined (const jerry_value_t value) /**< api value */
+{
+  return ecma_is_value_undefined (value);
+} /* jerry_value_is_undefined */
+
+/**
+ * Get boolean from the specified value.
  *
- * @return pointer to created array object
+ * @return true or false.
  */
-jerry_object_t *
-jerry_create_array_object (jerry_size_t size) /**< size of array */
+bool
+jerry_get_boolean_value (const jerry_value_t value) /**< api value */
 {
-  JERRY_ASSERT (size > 0);
+  JERRY_ASSERT (jerry_value_is_boolean (value));
```

Comment:
We could return with false if the value is not object to be fail-safe. Or that might be a bad idea, I don't know.